dataRetriever/resolvers/trieNodeResolver.go

```diff
@@ -208,6 +218,40 @@ func (tnRes *TrieNodeResolver) sendResponse(serializedNodes [][]byte, message p2
 	return tnRes.Send(buff, message.Peer())
 }
 
+func (tnRes *TrieNodeResolver) sendLargeMessage(
+	largeBuff []byte,
+	reference []byte,
+	chunkIndex int,
+	message p2p.MessageP2P,
+) error {
+
+	log.Warn("assembling chunk", "reference", reference, "len", len(largeBuff))
+	maxChunks := len(largeBuff) / maxBuffToSendTrieNodes
+	if len(largeBuff)%maxBuffToSendTrieNodes != 0 {
+		maxChunks++
+	}
+	chunkIndexOutOfBounds := chunkIndex < 0 || chunkIndex > maxChunks
+	if chunkIndexOutOfBounds {
+		return nil
```

Comment:
is there no need to at least log.Debug here?

Reply:
no, can be considered an attack vector.

Reply:
ok